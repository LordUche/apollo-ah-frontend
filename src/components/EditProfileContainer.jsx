import React, {Component} from 'react';
import { connect } from 'react-redux';

import EditProfile from '../views/EditProfile';
import ProfileHeaderContainer from './ProfileHeaderContainer';
import Wrapper from '../views/reuse/Wrapper';
import {updateUserProfile} from '../actions/updateProfileAction';


/**
 * Container component for the EditProfile view
 * @export
 * @class EditProfileContainer
 * @extends {Component}
 */
  class EditProfileContainer extends Component{
     /**
     *Creates an instance of EditProfileContainer.
     * @param {object} props The props object
     * @memberof EditProfileContainer
     */
    constructor(props){
        super(props);
        this.handleUpdateProfile = this.handleUpdateProfile.bind(this);
    }

    handleUpdateProfile(event){
      event.preventDefault();
      const data = new FormData(event.target);
      const {updateProfile} = this.props;
      updateProfile(data);
    }
    
     /**
     * The render method
     * @returns {array} The resulting JSX object
     * @memberof EditProfileContainer
     */
    render(){
      const {isLoading, profile, errorData} = this.props;
        return (
          <Wrapper>
            <ProfileHeaderContainer activePage="PROFILE" />
            <EditProfile 
            handleUpdateProfile={this.handleUpdateProfile} 
            isLoading={isLoading} 
            profile={profile}
            errorData={errorData}
            />
          </Wrapper>
        );
    }
}

const mapStateToProps = state => ({
  isLoading: state.loading,
  profile: state.user,
  errorData: state.user.error,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: userData => dispatch(updateUserProfile(userData)),
});
export default connect(mapStateToProps, mapDispatchToProps)(EditProfileContainer);
