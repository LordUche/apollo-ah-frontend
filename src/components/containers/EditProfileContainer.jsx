import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';

import EditProfile from '../views/EditProfile';
import ProfileHeaderContainer from './ProfileHeaderContainer';
import { updateUserProfile } from '../../actions/profileAction';
import { UPDATING_PROFILE } from '../../actions/actionTypes';

/**
 * Container component for the EditProfile view
 * @export
 * @class EditProfileContainer
 * @extends {Component}
 */
class EditProfileContainer extends Component {
  /**
     *Creates an instance of EditProfileContainer.
     * @param {object} props The props object
     * @memberof EditProfileContainer
     */
  constructor(props) {
    super(props);
    this.handleUpdateProfile = this.handleUpdateProfile.bind(this);
  }

  /**
   * Updates the user profile
   * @param {object} event The event object
   * @memberof EditProfileContainer
   */
  handleUpdateProfile(event) {
    event.preventDefault();
    const data = new FormData(event.target);
    const { updateProfile } = this.props;
    updateProfile(data);
  }

  render() {
    const { isLoading, profile } = this.props;
    return (
      <>
        <ProfileHeaderContainer activePage="PROFILE" />
        <EditProfile
          handleUpdateProfile={this.handleUpdateProfile}
          isLoading={isLoading}
          profile={profile}
        />
      </>
    );
  }
}

EditProfileContainer.propTypes = {
  updateProfile: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  profile: PropTypes.shape({
    firstname: PropTypes.string.isRequired,
    lastname: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired,
    errorData: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};


const mapStateToProps = state => ({
  isLoading: state.loading.updatingProfile,
  profile: state.user,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: userData => dispatch(updateUserProfile(UPDATING_PROFILE, userData)),
});
export default connect(mapStateToProps, mapDispatchToProps)(EditProfileContainer);
