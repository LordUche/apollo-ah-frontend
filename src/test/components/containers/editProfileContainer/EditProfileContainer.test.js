import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';

import EditProfileContainer from '../../../../components/containers/EditProfileContainer';

const mockStore = configMockStore([thunk]);
configure({ adapter: new Adapter() });

const setup = () => {
  const initialState = {
    user: {
      firstname: 'John',
      lastname: 'Doe',
      bio: 'I love writing',
      username: 'johnny',
      image: 'http://linktoimage.jpg',
      token: 'encoded',
      errorData: []
    },
    loading: {
      updatingProfile: false,
      updatingProfileImage: false
    },
    updateProfile: jest.fn(),
  };

  const connectedWrapper = mount(
    <Provider store={mockStore(initialState)}>
      <BrowserRouter>
        <EditProfileContainer activePage="PROFILE" />
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

describe('<EditProfileContainer>', () => {
  it('should render all children components', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
