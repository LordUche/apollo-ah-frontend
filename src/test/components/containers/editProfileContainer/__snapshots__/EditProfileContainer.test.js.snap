// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfileContainer> should render all children components 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(EditProfileContainer)
        activePage="PROFILE"
      >
        <EditProfileContainer
          activePage="PROFILE"
          isLoading={false}
          profile={
            Object {
              "bio": "I love writing",
              "errorData": Array [],
              "firstname": "John",
              "image": "http://linktoimage.jpg",
              "lastname": "Doe",
              "token": "encoded",
              "username": "johnny",
            }
          }
          updateProfile={[Function]}
        >
          <Connect(ProfileHeaderContainer)
            activePage="PROFILE"
          >
            <ProfileHeaderContainer
              activePage="PROFILE"
              isLoading={false}
              profile={
                Object {
                  "bio": "I love writing",
                  "errorData": Array [],
                  "firstname": "John",
                  "image": "http://linktoimage.jpg",
                  "lastname": "Doe",
                  "token": "encoded",
                  "username": "johnny",
                }
              }
              showLoader={[Function]}
              updateProfile={[Function]}
            >
              <ProfileHeader
                activePage="PROFILE"
                handleImageUpload={[Function]}
                isLoading={false}
                profile={
                  Object {
                    "bio": "I love writing",
                    "errorData": Array [],
                    "firstname": "John",
                    "image": "http://linktoimage.jpg",
                    "lastname": "Doe",
                    "token": "encoded",
                    "username": "johnny",
                  }
                }
              >
                <header
                  className="page-header"
                >
                  <h1>
                    My PROFILE
                  </h1>
                </header>
                <div
                  className="profile-box mt-n4"
                >
                  <div
                    className="d-inline-block relative profile-img-cover"
                  >
                    <span
                      className="spinner-border text-light align-middle"
                    />
                    <img
                      alt="johnny"
                      className="img-thumbnail"
                      src="http://linktoimage.jpg"
                    />
                    <button
                      className="btn change-image transition"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="fas fa-camera"
                      />
                    </button>
                  </div>
                  <div
                    className="profile-wrapper"
                  >
                    <h4
                      className="profile-name text-capitalize"
                    >
                      John
                       
                      Doe
                    </h4>
                    <p
                      className="profile-username"
                    >
                      @johnny
                    </p>
                  </div>
                  <div
                    className="profile-footer"
                  >
                    <Link
                      className="btn false"
                      replace={false}
                      to="dashboard"
                    >
                      <a
                        className="btn false"
                        href="/dashboard"
                        onClick={[Function]}
                      >
                        <span
                          className="relative"
                        >
                          Dashboard
                        </span>
                      </a>
                    </Link>
                    <Link
                      className="btn active"
                      replace={false}
                      to="profile"
                    >
                      <a
                        className="btn active"
                        href="/profile"
                        onClick={[Function]}
                      >
                        <span
                          className="relative"
                        >
                          Edit profile
                        </span>
                      </a>
                    </Link>
                    <Link
                      className="btn false"
                      replace={false}
                      to="settings"
                    >
                      <a
                        className="btn false"
                        href="/settings"
                        onClick={[Function]}
                      >
                        <span
                          className="relative"
                        >
                          Settings
                        </span>
                      </a>
                    </Link>
                  </div>
                </div>
              </ProfileHeader>
            </ProfileHeaderContainer>
          </Connect(ProfileHeaderContainer)>
          <EditProfile
            handleUpdateProfile={[Function]}
            isLoading={false}
            profile={
              Object {
                "bio": "I love writing",
                "errorData": Array [],
                "firstname": "John",
                "image": "http://linktoimage.jpg",
                "lastname": "Doe",
                "token": "encoded",
                "username": "johnny",
              }
            }
          >
            <div
              className="card mb-5 mt-4"
            >
              <form
                className="form p-sm-4 p-3 "
                noValidate={true}
                onSubmit={[Function]}
              >
                <div
                  className="form-group mb-4"
                >
                  <label
                    className="font-weight-bold"
                    htmlFor="fname"
                  >
                    First Name
                  </label>
                  <input
                    className="form-control text-capitalize"
                    defaultValue="John"
                    id="fname"
                    name="firstname"
                    required={true}
                    type="text"
                  />
                  <span
                    className="text-danger"
                  >
                    First name is required.
                  </span>
                </div>
                <div
                  className="form-group mb-4"
                >
                  <label
                    className="font-weight-bold"
                    htmlFor="lname"
                  >
                    Last Name
                  </label>
                  <input
                    className="form-control text-capitalize"
                    defaultValue="Doe"
                    id="lname"
                    name="lastname"
                    required={true}
                    type="text"
                  />
                  <span
                    className="text-danger"
                  >
                    Last name is required.
                  </span>
                </div>
                <div
                  className="form-group mb-4"
                >
                  <label
                    className="font-weight-bold"
                    htmlFor="fname"
                  >
                    Username
                  </label>
                  <input
                    className="form-control"
                    defaultValue="johnny"
                    id="username"
                    name="username"
                    required={true}
                    type="text"
                  />
                  <span
                    className="text-danger"
                  >
                    Username is required.
                  </span>
                </div>
                <div
                  className="form-group mb-4"
                >
                  <label
                    className="font-weight-bold"
                    htmlFor="bio"
                  >
                    Bio
                  </label>
                  <textarea
                    className="form-control"
                    defaultValue="I love writing"
                    id="bio"
                    name="bio"
                    required={true}
                  />
                  <span
                    className="text-danger"
                  >
                    Bio is required.
                  </span>
                </div>
                <button
                  className="btn btn-brand btn-plain"
                  disabled={false}
                  type="submit"
                >
                  Update profile
                   
                </button>
                <input
                  className="hidden"
                  defaultValue="encoded"
                  name="token"
                  type="hidden"
                />
                <input
                  className="hidden"
                  defaultValue="http://linktoimage.jpg"
                  name="image"
                  type="hidden"
                />
              </form>
            </div>
          </EditProfile>
        </EditProfileContainer>
      </Connect(EditProfileContainer)>
    </Router>
  </BrowserRouter>
</Provider>
`;
